doc/usage/extensions/autosummary.rst

```diff
@@ -136,13 +141,20 @@ also use these config values:
 .. confval:: autosummary_generate
 
    Boolean indicating whether to scan all found documents for autosummary
-   directives, and to generate stub pages for each.
+   directives, and to generate stub pages for each. It is disabled by default.
 
    Can also be a list of documents for which stub pages should be generated.
 
    The new files will be placed in the directories specified in the
    ``:toctree:`` options of the directives.
 
+.. confval:: autosummary_recursive
```

Comment:
It would be also good to add `:recursive:` option to `autosummary` directive. It allows to switch recursive-ness for each autosummary calls. What do you think?

Reply:
I don't know what to do without discussing based on the use case.

* What kind of cases do you want to control recursion with common settings?
* What kind of cases do you want to control recursion separately for the autosummary directive?

Reply:
@tk0miya Would this require adding something like `sphinx/ext/autodoc/directive.py` to the autosummary extension? I'm not familiar with this aspect but I could try to figure it out.

@shimizukawa This would allow the user to exclude specific packages from being explored when `autosummary_recursive == True`. To work it requires `autosummary_overwrite == False` and the user adding for example `package.excludepackage.rst`. The `excludepackage` package itself would be in the docs though, so I'm not sure what the use case is for this.

Reply:
As a trial, I implemented :recursive: option. Please see https://github.com/tk0miya/sphinx/tree/6040_autosummary_recursive.